api/src/main/java/com/epam/pipeline/manager/cluster/autoscale/AutoscaleManager.java

```diff
@@ -477,6 +480,7 @@ private void createNodeForRun(List<CompletableFuture<Void>> tasks, String runId,
                                       InstanceRequest requiredInstance) {
             long longId = Long.parseLong(runId);
             addNodeUpTask(longId);
+            setAvailabilityZoneIfSpecified(requiredInstance, longId);
```

Comment:
Maybe we can set AZ in method getNewRunInstance where PipelineRun is already loaded instead of setting it separately?

Reply:
Done